src/re.rs

```diff
@@ -1025,19 +1044,16 @@ impl<'t> Iterator for SubCapturesPos<'t> {
 /// name and the value.
 ///
 /// `'t` is the lifetime of the matched text.
-pub struct SubCapturesNamed<'t>{
-    caps: &'t Captures<'t>,
-    inner: Option<Iter<'t, String, usize>>,
+pub struct SubCapturesNamed<'c, 't: 'c> {
+    caps: &'c Captures<'t>,
+    names: Box<Iterator<Item=(&'c str, usize)> + 'c>,
```

Comment:
Hmm, the boxed iterator here is unfortunate. To avoid it, could we split the iterator out into a separate type? It will probably need to be an enum itself. (I know it's a pain, sorry.)